src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -1365,6 +1365,68 @@ public IActionResult SyncFromDate([FromBody] WalletSyncFromDateRequest request)
             return this.Ok();
         }
 
+        [Route("wallet-stats")]
+        [HttpGet]
+        public IActionResult WalletStats([FromQuery] WalletStatsRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+            WalletStatsModel model = new WalletStatsModel();
+            model.WalletName = request.WalletName;
+
+            // Checks the request is valid.
+            if (!this.ModelState.IsValid)
+            {
+                return ModelStateErrors.BuildErrorResponse(this.ModelState);
+            }
+
+            try
+            {
+                IEnumerable<UnspentOutputReference> spendableTransactions = this.walletManager.GetSpendableTransactionsInAccount(new WalletAccountReference(request.WalletName, request.AccountName), request.MinConfirmations);
+                
+                model.TotalUtxoCount = spendableTransactions.Count();
+                model.UniqueTransactionCount = (from s in spendableTransactions
```

Comment:
@maciejzaleski we don't really do LINQ queries like this anywhere in the solution, so we tend to stick LINQ like so:

```
model.UniqueTransactionCount = spendableTransactions.GroupBy(s=>s.Transaction.Id).Count();
```

Reply:
I reject this comment. The code works and is readable.